packages/test/src/test-integration.ts

```diff
@@ -773,4 +775,120 @@ if (RUN_INTEGRATION_TESTS) {
       iface.temporal.api.enums.v1.WorkflowExecutionStatus.WORKFLOW_EXECUTION_STATUS_FAILED
     );
   });
+
+  test('WorkflowClient.start fails with WorkflowExecutionAlreadyStartedError', async (t) => {
+    const { client } = t.context;
+    const workflowId = uuid4();
+    const handle = await client.start(workflows.sleeper, { taskQueue: 'test', workflowId, args: [10000000] });
+    try {
+      await t.throwsAsync(
+        client.start(workflows.sleeper, {
+          taskQueue: 'test',
+          workflowId,
+        }),
+        { instanceOf: WorkflowExecutionAlreadyStartedError, message: 'Workflow execution already started' }
+      );
+    } finally {
+      await handle.terminate();
+    }
+  });
+
+  test('Handle from WorkflowClient.start follows only own execution chain', async (t) => {
+    const { client } = t.context;
+    const workflowId = uuid4();
+    const handleFromThrowerStart = await client.start(workflows.throwAsync, { taskQueue: 'test', workflowId });
+    const handleFromGet = client.getHandle(workflowId);
+    await t.throwsAsync(handleFromGet.result(), { message: /.*/ });
+    const handleFromSleeperStart = await client.start(workflows.sleeper, {
+      taskQueue: 'test',
+      workflowId,
+      args: [1_000_000],
+    });
+    try {
+      await t.throwsAsync(handleFromThrowerStart.result(), { message: 'Workflow execution failed' });
+    } finally {
+      await handleFromSleeperStart.terminate();
+    }
+  });
+
+  test('Handle from WorkflowClient.signalWithStart follows only own execution chain', async (t) => {
+    const { client } = t.context;
+    const workflowId = uuid4();
+    const handleFromThrowerStart = await client.signalWithStart(workflows.throwAsync, {
+      taskQueue: 'test',
+      workflowId,
+      signal: 'unblock',
+      signalArgs: [],
+    });
+    const handleFromGet = client.getHandle(workflowId);
+    await t.throwsAsync(handleFromGet.result(), { message: /.*/ });
+    const handleFromSleeperStart = await client.start(workflows.sleeper, {
+      taskQueue: 'test',
+      workflowId,
+      args: [1_000_000],
+    });
+    try {
+      await t.throwsAsync(handleFromThrowerStart.result(), { message: 'Workflow execution failed' });
+    } finally {
+      await handleFromSleeperStart.terminate();
+    }
+  });
+
+  test('Handle from WorkflowClient.getHandle follows only own execution chain', async (t) => {
+    const { client } = t.context;
+    const workflowId = uuid4();
+    const handleFromThrowerStart = await client.start(workflows.throwAsync, {
+      taskQueue: 'test',
+      workflowId,
+    });
+    const handleFromGet = client.getHandle(workflowId, undefined, {
+      firstExecutionRunId: handleFromThrowerStart.originalRunId,
```

Comment:
when should we tell people to use this in docs?

Reply:
When they want to make sure that they don't terminate / cancel a workflow if it's not part of a specific execution chain.